extensions/schema-compare/src/utils.ts

```diff
@@ -74,17 +84,44 @@ function connectionInfoToConnectionProfile(details: azdata.ConnectionInfo): azda
 	};
 }
 
-export async function verifyConnectionAndGetOwnerUri(endpoint: mssql.SchemaCompareEndpointInfo): Promise<string> {
+export async function verifyConnectionAndGetOwnerUri(endpoint: mssql.SchemaCompareEndpointInfo, caller: string): Promise<string> {
+	let ownerUri = undefined;
 	if (endpoint.endpointType === mssql.SchemaCompareEndpointType.Database && endpoint.connectionDetails) {
-		let connection = await azdata.connection.connect(connectionInfoToConnectionProfile(endpoint.connectionDetails), false, false);
+		let connectionProfile = await connectionInfoToConnectionProfile(endpoint.connectionDetails);
+		let connection = await azdata.connection.connect(connectionProfile, false, false);
+		if (connection) {
+			ownerUri = await azdata.connection.getUriForConnection(connection.connectionId);
+			if (!ownerUri) {
+				let connectionList = await azdata.connection.getConnections(true);
+				let userConnection;
+				userConnection = connectionList.find(connection =>
+					(endpoint.connectionDetails['authenticationType'] === 'SqlLogin'
+						&& endpoint.connectionDetails['serverName'] === connection.options.server
+						&& endpoint.connectionDetails['userName'] === connection.options.user
+						&& (endpoint.connectionDetails['databaseName'].toLowerCase() === connection.options.database.toLowerCase()
+							|| connection.options.database.toLowerCase() === 'master')));
+
+				if (userConnection === undefined) {
+					const getConnectionString = localize('schemaCompare.GetConnectionString', "Do you want to connect to {0}?", caller);
+					// need only yes button - since the modal dialog has a default cancel
+					const yesString = localize('schemaCompare.ApplyYes', "Yes");
+
+					let result = await vscode.window.showWarningMessage(getConnectionString, { modal: true }, yesString);
+					if (result === yesString) {
+						userConnection = await azdata.connection.openConnectionDialog(null, connectionProfile);
```

Comment:
We generally prefer undefined over null

Reply:
Don't they have different intentions?  undefined being not initialized, null being the value is unavailable/missing (as would be here, if `connectionList.find` doesn't yield anything).  If we're following that model, then `null` feels more appropriate.

Reply:
The first argument is an array of Providers, if that makes a difference. 
At other points in code, "openConnectionDialog" is called with undefined instead of null. By virtue of that, I will modify it to say undefined.
Let me know otherwise.

Reply:
@Benjin Yes, your understanding of null vs undefined is correct. But we just don't use that model in our code base currently - it's all undefineds.

Reply:
Good point.. I do see that in code. Anyone knows why?
+ @anthonydresser  :)

Reply:
While what @Benjin said is correct, its all just semantics. But neither is "right" from a functionality standpoint (the engine doesn't enforce anything). So rather than try and think "in this scenario what is the correct usage, null or undefined" and running into issues with people not coming to different conclusions and running into issues, its easier to just always using 1, and vscode chose undefined, so thats what we do. 🤷‍♂ It just makes things easier.